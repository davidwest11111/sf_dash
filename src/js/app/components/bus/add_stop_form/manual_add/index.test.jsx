import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import ManualAdd from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let onAddOrRemoveStopSpy;

  beforeEach(() => {
    onAddOrRemoveStopSpy = jest.fn();
  });

  test('(default)', () => {
    wrapper = shallow((
      <ManualAdd onAddOrRemoveStop={onAddOrRemoveStopSpy} />
    ));
  });

  test('stop added was invalid', () => {
    wrapper = shallow((
      <ManualAdd onAddOrRemoveStop={onAddOrRemoveStopSpy} />
    ));
    wrapper.setState({
      stopIsValid: false,
    });
  });

  test('stop added was valid', () => {
    wrapper = shallow((
      <ManualAdd onAddOrRemoveStop={onAddOrRemoveStopSpy} />
    ));
    wrapper.setState({
      stopIsValid: true,
    });
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
