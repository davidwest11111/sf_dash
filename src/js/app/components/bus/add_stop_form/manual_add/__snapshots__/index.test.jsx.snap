// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `
<Form
  className=""
  onSubmit={[Function]}
  tag="form"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="manual-add-route-tag"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Route Tag:
        </Label>
        <Input
          className=""
          id="manual-add-route-tag"
          onChange={[Function]}
          placeholder="12"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="manual-add-route-stop"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Stop Tag:
        </Label>
        <Input
          className=""
          id="manual-add-route-stop"
          onChange={[Function]}
          placeholder="5837"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
  </Row>
  <div>
    <Button
      color="primary"
      tag="button"
      type="submit"
    >
      Validate & Add Stop
    </Button>
  </div>
</Form>
`;

exports[`outputs the expected tree when stop added was invalid 1`] = `
<Form
  className=""
  onSubmit={[Function]}
  tag="form"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="manual-add-route-tag"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Route Tag:
        </Label>
        <Input
          className="is-invalid"
          id="manual-add-route-tag"
          onChange={[Function]}
          placeholder="12"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="manual-add-route-stop"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Stop Tag:
        </Label>
        <Input
          className="is-invalid"
          id="manual-add-route-stop"
          onChange={[Function]}
          placeholder="5837"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
  </Row>
  <div>
    <Button
      color="primary"
      tag="button"
      type="submit"
    >
      Validate & Add Stop
    </Button>
  </div>
</Form>
`;

exports[`outputs the expected tree when stop added was valid 1`] = `
<Form
  className=""
  onSubmit={[Function]}
  tag="form"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="manual-add-route-tag"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Route Tag:
        </Label>
        <Input
          className="is-valid"
          id="manual-add-route-tag"
          onChange={[Function]}
          placeholder="12"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="6"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="manual-add-route-stop"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Stop Tag:
        </Label>
        <Input
          className="is-valid"
          id="manual-add-route-stop"
          onChange={[Function]}
          placeholder="5837"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
  </Row>
  <div>
    <Button
      color="primary"
      tag="button"
      type="submit"
    >
      Validate & Add Stop
    </Button>
  </div>
</Form>
`;
